feat: update the vault bdd requirements to use external-dns and TLS

ccojocar opened the pull request.

ccojocar committed:
feat: update the vault bdd requirements to use external-dns and TLS

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
chore: update the builder-go to version 0.1.704

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

rawlingsj commented on bdd/boot-vault/jx-requirements.yml:
```diff
@@ -14,13 +14,15 @@ environments:
-    production: false
+    email: jenkins-x@googlegroups.com
+    enabled: true
+    production: true
```
I'm not sure we should be using the production lets encrypt service during testing, we will likely be rate limited via our domain or email address.  For testing they recommend using the staging service.

ccojocar replied:
We cannot really test E2E with certificates issued by staging let's encrypt service. They are self-singed and various clients fail along the way because the certificate is invalid. I have already tried this but it doesn't work.

The rate limit seems to be applied on `*.bdd.jenkins-x.rocks` TLD. I am thinking that we can register 10-15 TLDs (e.g. bdd1.jenkins-x.rocks, bdd2.jenkins-x.rocks...) and rotate through them every time a BDD test is started. What's your opinion on this?

pmuir replied:
Makes sense to me.

ccojocar replied:
I registered 12 DNS managed zones and rotate them each hour. Let's see how it works.

ccojocar committed:
chore: remove the execute permissions from some files

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
chore: increase the version of builder-go to 0.1.728

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
feat: implement a domain rotation for vault BDD

The domain is rotated every hour based on a simple scheme such as `domain_prefix{hour}.domain_suffix`. There are
12 DNS managed zones with this name created in the jenkins-x-bdd2 project on GCP.

This trick tries to prevent the cert-manager to reach the lets encrypt API rate limits when running the BDD tests.

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
chore: disable TLS for staging and production environments

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
fix: use the correct path for jx-requirements when rotating the domain

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
chore: upgrade builders to version 0.1.730

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar commented:
/test bdd-vault

ccojocar commented:
/test bdd-vault

ccojocar committed:
chore: update builders to version 0.1.731

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

abayer commented:
/lgtm

ccojocar committed:
chore: use latest tag from version stream

ccojocar committed:
chore: use 0.1.741 builder version for testing

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

jstrachan commented:
/lgtm
